scanner.go

```diff
@@ -51,29 +67,30 @@ func (s *scanner) fetch() ([]*pb.Result, error) {
 		}
 
 		if rs := resp.Results; len(rs) > 0 {
-			return rs, nil
+			return rs, sm, nil
 		} else if s.closed {
-			return nil, io.EOF
+			return nil, nil, io.EOF
 		}
 	}
 }
 
-func (s *scanner) peek() (*pb.Result, error) {
+func (s *scanner) peek() (*pb.Result, *hrpc.ScanMetrics, error) {
 	if len(s.results) == 0 {
 		if s.closed {
 			// done scanning
-			return nil, io.EOF
+			return nil, nil, io.EOF
 		}
 
-		rs, err := s.fetch()
+		rs, sm, err := s.fetch()
 		if err != nil {
-			return nil, err
+			return nil, nil, err
 		}
 
 		// fetch cannot return zero results
 		s.results = rs
+		s.metrics = sm
```

Comment:
Not sure s.metrics should be altered by peek(), maybe just return sm and not override s.metrics?

And a general question, should s.metrics ever be allowed to be changed inside a call of type func(s *scanner)?